exporter/containerimage/exptypes/types.go

```diff
@@ -24,3 +25,29 @@ type Platform struct {
 	ID       string
 	Platform ocispecs.Platform
 }
+
+// BuildInfo defines build dependencies that will be added to image config as
+// moby.buildkit.buildinfo.v0 key and returned in solver ExporterResponse as
+// ExporterBuildInfo key.
+type BuildInfo struct {
+	// Type defines the BuildInfoType source type (docker-image, git, http).
+	Type BuildInfoType `json:"type,omitempty"`
+	// Ref is the reference of the source.
+	Ref string `json:"ref,omitempty"`
+	// Alias is a special field used to match with the actual source ref
+	// because frontend might have already transformed a string user typed
+	// before generating LLB.
+	Alias string `json:"alias,omitempty"`
+	// Pin is the source digest.
+	Pin string `json:"pin,omitempty"`
+}
+
+// BuildInfoType contains source type.
+type BuildInfoType string
+
+// List of source types.
+const (
+	BuildInfoTypeDockerImage BuildInfoType = "docker-image"
```

Comment:
These are already defined in `source`. If there is an importing issue maybe it makes sense to move them but ideally, they are the same constants.

Reply:
moved to `source/types`